### PR TITLE
refactoring names, some traits, added pi-client

### PR DESCRIPTION
**Pull request by tamaspolgar**
what do you think?
- I added ??? as default impl to the traits,
- I renamed the many GpioImpl to corresponding XYZGpio,
- there was no real Pi client, I created it.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -52,8 +52,15 @@ lazy val remoteServer = Project("remote-server", file("remote-server"))
   .settings(Testing.settings: _*)
   .settings(Assembly.remoteServerAssemblySettings: _*)
 
+lazy val piClient = Project("pi-client", file("pi-client"))
```
Comment (lachatak) [inline build.sbt]: is this really needed? just lets the request flow through. I suppose it is not needed at all.
he?

File: console-client/src/main/scala/org/kaloz/pi4j/client/console/ConsoleClientFactory.scala [hunk 2]
```diff
@@ -9,9 +9,9 @@ class ConsoleClientFactory extends ClientFactory {
 
   private lazy val stubClientActor = system.actorOf(StubClientActor.props)
 
-  lazy val gpio = new GpioImpl(stubClientActor)
-  lazy val gpioUtil = new GpioUtilImpl(stubClientActor)
-  lazy val gpioInterrupt = new GpioInterruptImpl(stubClientActor)
+  lazy val gpio = new ConsoleGpio(stubClientActor)
```
Comment (lachatak) [inline console-client/src/main/scala/org/kaloz/pi4j/client/console/ConsoleClientFactory.scala:12]: :+1:

File: console-client/src/main/scala/org/kaloz/pi4j/client/console/package.scala [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package org.kaloz.pi4j.client
+
+import akka.util.Timeout
+import scala.concurrent.duration._
+
+package object console {
+
+  implicit val timeout = Timeout(1 minutes)
```
Comment (lachatak) [inline console-client/src/main/scala/org/kaloz/pi4j/client/console/package.scala]: Why it is better here?

File: pi-client/src/main/scala/org/kaloz/pi4j/client/pi/PiGpio.scala [hunk 4]
```diff
@@ -0,0 +1,63 @@
+package org.kaloz.pi4j.client.pi
+
+import com.pi4j.wiringpi.{Gpio => JGpio, GpioInterruptCallback}
+import org.kaloz.pi4j.client.Gpio
+
+class PiGpio extends Gpio {
```
Comment (lachatak) [inline pi-client/src/main/scala/org/kaloz/pi4j/client/pi/PiGpio.scala]: as I asked. is this needed?
just omit the weaving and you will got the same.
probably it not worth the hassle to maintain..

File: project/Assembly.scala [hunk 5]
```diff
@@ -40,6 +40,11 @@ object Assembly {
       assemblyJarName in assembly := "pi4j-remote-server.jar"
     )
 
+  lazy val piClientAssemblySettings =
```
Comment (lachatak) [inline project/Assembly.scala]: same

File: project/Dependencies.scala [hunk 6]
```diff
@@ -107,6 +107,14 @@ object Dependencies {
     scalaTest     	% "test"
   )
 
+  val pi = deps(
```
Comment (lachatak) [inline project/Dependencies.scala]: same

**Comment by lachatak** [comment]
done. Apart from the pi stuff it is ok.

File: build.sbt (hunk 1 above)
Comment (tamaspolgar) [inline build.sbt]: ok, makes sense. I'll remove it

File: pi-client/src/main/scala/org/kaloz/pi4j/client/pi/PiGpio.scala (hunk 4 above)
Comment (tamaspolgar) [inline pi-client/src/main/scala/org/kaloz/pi4j/client/pi/PiGpio.scala]: ok, makes sense. I'll remove it

File: console-client/src/main/scala/org/kaloz/pi4j/client/console/package.scala (hunk 3 above)
Comment (tamaspolgar) [inline console-client/src/main/scala/org/kaloz/pi4j/client/console/package.scala]: it's not really better.. I reverted it

**Comment by tamaspolgar** [comment]
check it now

**Pull request merged**